cli/azd/pkg/infra/scope.go

```diff
@@ -14,126 +16,181 @@ import (
+	// Deployment fetches information about this deployment.
+	Deployment(ctx context.Context) (*armresources.DeploymentExtended, error)
+	// Operations returns all the operations for this deployment.
+	Operations(ctx context.Context) ([]*armresources.DeploymentOperation, error)
 }
 
-type ResourceGroupScope struct {
-	azCli          azcli.AzCli
-	name           string
-	subscriptionId string
-	resourceGroup  string
+type ResourceGroupScopedDeployment struct {
+	*ResourceGroupScope
+	name string
 }
 
-// Gets the Azure subscription id
-func (s *ResourceGroupScope) SubscriptionId() string {
-	return s.subscriptionId
+func (s *ResourceGroupScopedDeployment) Name() string {
+	return s.name
 }
 
-// Gets the deployment name
-func (s *ResourceGroupScope) Name() string {
-	return s.name
+// Gets the Azure subscription id
+func (s *ResourceGroupScopedDeployment) SubscriptionId() string {
+	return s.subscriptionId
 }
 
 // Gets the resource group name
-func (s *ResourceGroupScope) ResourceGroup() string {
-	return s.resourceGroup
+func (s *ResourceGroupScopedDeployment) ResourceGroupName() string {
+	return s.resourceGroupName
 }
 
-func (s *ResourceGroupScope) Deploy(
-	ctx context.Context, template azure.RawArmTemplate, parameters azure.ArmParameters,
+func (s *ResourceGroupScopedDeployment) Deploy(
+	ctx context.Context, template azure.RawArmTemplate, parameters azure.ArmParameters, tags map[string]*string,
 ) (*armresources.DeploymentExtended, error) {
-	return s.azCli.DeployToResourceGroup(ctx, s.subscriptionId, s.resourceGroup, s.name, template, parameters)
+	return s.azCli.DeployToResourceGroup(ctx, s.subscriptionId, s.resourceGroupName, s.name, template, parameters, tags)
 }
 
 // GetDeployment fetches the result of the most recent deployment.
-func (s *ResourceGroupScope) GetDeployment(ctx context.Context) (*armresources.DeploymentExtended, error) {
-	return s.azCli.GetResourceGroupDeployment(ctx, s.subscriptionId, s.resourceGroup, s.name)
+func (s *ResourceGroupScopedDeployment) Deployment(ctx context.Context) (*armresources.DeploymentExtended, error) {
+	return s.azCli.GetResourceGroupDeployment(ctx, s.subscriptionId, s.resourceGroupName, s.name)
 }
 
 // Gets the resource deployment operations for the current scope
-func (s *ResourceGroupScope) GetResourceOperations(ctx context.Context) ([]*armresources.DeploymentOperation, error) {
-	return s.azCli.ListResourceGroupDeploymentOperations(ctx, s.subscriptionId, s.resourceGroup, s.name)
+func (s *ResourceGroupScopedDeployment) Operations(ctx context.Context) ([]*armresources.DeploymentOperation, error) {
+	return s.azCli.ListResourceGroupDeploymentOperations(ctx, s.subscriptionId, s.resourceGroupName, s.name)
 }
 
 // Gets the url to check deployment progress
-func (s *ResourceGroupScope) DeploymentUrl() string {
-	return azure.ResourceGroupDeploymentRID(s.subscriptionId, s.resourceGroup, s.name)
+func (s *ResourceGroupScopedDeployment) PortalUrl() string {
+	return fmt.Sprintf("%s/%s",
```

Comment:
```suggestion
	return fmt.Sprintf("%s%s",
```

Reply:
Noticed this while trying to click on a link while playing around with this.  The prefix contained the trailing slash.